lib/shared-functions.sh

```diff
@@ -19,6 +19,22 @@ function read_image_version() {
   IMAGE_VERSION_PATCH=${BASH_REMATCH[3]}
 }
 
+function read_mongo_version() {
+  local mongo_image
+  mongo_image=$(read_configuration "MONGO_IMAGE")
+  if [[ ! "$mongo_image" =~ ^mongo:([0-9]+)\.(.*)$ ]]; then
```

Comment:
I'm aware of customers with custom docker registries. Their images would not match this regex. WDYT about removing the name and just matching the tag?

As an alternative, we could split the `MONGO_IMAGE` variable into two parts, the name and the version.

Reply:
I added a check for `MONGO_IMAGE`:

- If `MONGO_VERSION` is provided it reads the major version from there.
- If not provided, checks whether `MONGO_IMAGE` is a stock version and parses the major version.
- If `MONGO_IMAGE` is not a stock image, issues a warning and defaults to `5` for backwards compatibility.

The warning might be printed more than once (tried to add a check for `MONGOSH` existence check, but the variables are unbounded in subshells), which I think we can live with.